tests/Unit/Evolution/Systems/GeneralizedHarmonic/Test_CharFields.cpp

```diff
@@ -0,0 +1,138 @@
+          Frame>,
+      "TestFunctionsCharFields", "char_field_uplus", {{{-100.0, 100.0}}},
+      used_for_size);
+  // UMinus
+  pypp::check_with_random_values<1>(
+      compute_field_with_tag<
+          ::Tags::Field<GeneralizedHarmonic::Tags::UMinus<Dim, Frame>>, Dim,
+          Frame>,
+      "TestFunctionsCharFields", "char_field_uminus", {{{-100.0, 100.0}}},
+      used_for_size);
+}
+}  // namespace
+
+
+namespace {
+template <typename Tag, size_t Dim, typename Frame>
+typename Tag::type compute_evol_field_with_tag(
+    const Scalar<DataVector>& gamma_2,
+    const tnsr::aa<DataVector, Dim, Frame>& u_psi,
+    const tnsr::iaa<DataVector, Dim, Frame>& u_zero,
+    const tnsr::aa<DataVector, Dim, Frame>& u_plus,
+    const tnsr::aa<DataVector, Dim, Frame>& u_minus,
+    const tnsr::i<DataVector, Dim, Frame>& normal_one_form) {
+  return get<Tag>(
+      GeneralizedHarmonic::EvolvedFieldsFromCharacteristicCompute<Dim,
+          Frame>::function(
+              gamma_2, u_psi, u_zero, u_plus, u_minus, normal_one_form));
+}
+
+template <size_t Dim, typename Frame>
+void test_evolved_from_characteristic_fields() noexcept {
+  const DataVector used_for_size(20);
+  // Psi
+  pypp::check_with_random_values<1>(
+      compute_evol_field_with_tag<
+          ::Tags::Field<GeneralizedHarmonic::Tags::Psi<Dim, Frame>>,
+          Dim, Frame>,
+      "TestFunctionsCharFields", "evol_field_psi", {{{-100.0, 100.0}}},
+      used_for_size);
+  // Pi
+  pypp::check_with_random_values<1>(
+      compute_evol_field_with_tag<
+          ::Tags::Field<GeneralizedHarmonic::Tags::Pi<Dim, Frame>>, Dim,
+          Frame>,
+      "TestFunctionsCharFields", "evol_field_pi", {{{-100.0, 100.0}}},
+      used_for_size);
+  // Phi
+  pypp::check_with_random_values<1>(
+      compute_evol_field_with_tag<
+          ::Tags::Field<GeneralizedHarmonic::Tags::Phi<Dim, Frame>>, Dim,
+          Frame>,
+      "TestFunctionsCharFields", "evol_field_phi", {{{-100.0, 100.0}}},
+      used_for_size);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Evolution.Systems.GeneralizedHarmonic.CharFields",
+                  "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/GeneralizedHarmonic/"};
+
+  test_characteristic_fields<1, Frame::Grid>();
+  test_characteristic_fields<2, Frame::Grid>();
+  test_characteristic_fields<3, Frame::Grid>();
+  test_characteristic_fields<1, Frame::Inertial>();
+  test_characteristic_fields<2, Frame::Inertial>();
+  test_characteristic_fields<3, Frame::Inertial>();
+
+  test_evolved_from_characteristic_fields<1, Frame::Grid>();
+  test_evolved_from_characteristic_fields<2, Frame::Grid>();
+  test_evolved_from_characteristic_fields<3, Frame::Grid>();
+  test_evolved_from_characteristic_fields<1, Frame::Inertial>();
+  test_evolved_from_characteristic_fields<2, Frame::Inertial>();
+  test_evolved_from_characteristic_fields<3, Frame::Inertial>();
+}
```

Comment:
Add a test that computes the char fields and char speeds for KerrSchild. If the algebra is too tedious for generic spin, testing only spin=0 is fine with me.

Reply:
see reply to https://github.com/sxs-collaboration/spectre/pull/1129#discussion_r229855687